homeassistant/util/package.py

```diff
@@ -59,12 +65,43 @@ def check_package_exists(package: str, lib_dir: str) -> bool:
         # This is a zip file
         req = pkg_resources.Requirement.parse(urlparse(package).fragment)
 
-    # Check packages from lib dir
-    if lib_dir is not None:
-        if any(dist in req for dist in
-               pkg_resources.find_distributions(lib_dir)):
-            return True
+    env = pkg_resources.Environment()
+    return any(dist in req for dist in env[req.project_name])
+
+
+def is_virtual_env() -> bool:
+    """Return true if environment is a virtual environment."""
+    return hasattr(sys, 'real_prefix')
 
-    # Check packages from global + virtual environment
-    # pylint: disable=not-an-iterable
-    return any(dist in req for dist in pkg_resources.working_set)
+
+def _get_user_site(deps_dir: str) -> tuple:
+    """Get arguments and environment for subprocess used in get_user_site."""
+    env = os.environ.copy()
+    env['PYTHONUSERBASE'] = os.path.abspath(deps_dir)
+    args = [sys.executable, '-m', 'site', '--user-site']
+    return args, env
+
+
+def get_user_site(deps_dir: str) -> str:
```

Comment:
It's used in `bootstrap.py` in the sync version of `mount_local_lib_path`  which in turn is used by the `scripts` package. Let me know if I should change that.